CONTRIBUTING.md

```diff
@@ -51,5 +51,104 @@ This process can be performed automatically by running `make license-generator`.
 
 Sometimes pull requests take a little while to be merged. This is partially because they often have knock-on effects for the rest of the ecosystem, and partially because we want to give core team members time to review and consider changes thoroughly. Please see the organization's [governance document](https://github.com/purescript/governance) for information about when a pull request may be merged.
 
+## Developer Guide
+
+The following instructions are intended to help PureScript users more easily contribute to the compiler, even if this is your first Haskell project.
+
+### Prerequisites
+
+Install `stack`. [Instructions](https://docs.haskellstack.org/en/stable/README/).
+
+See the [PureScript Install Guide](INSTALL.md) for information on the `curses`/`ncurses` dependency.
+
+Update stack's package index before proceeding:
+```
+stack update
+```
+
+### Clone
+
+```
+git clone https://github.com/purescript/purescript.git purescript_compiler
+cd purescript_compiler
+```
+
+### Build
+
+```
+stack build
+```
+
+This will take a while the first time it is run.
+
+### Install
+
+If you'd like to test the build in a local shell, you can run:
+```
+stack exec bash
+```
+
+This prepends the location of the compiler to your `$PATH`, so you can test your changes. This works with other tools that depend on the `purs` binary, such as `spago`. Note that rebuilds continue to use this same path, so you don't need to keep re-running `stack exec` to test new changes (assuming you're using a separate terminal for `stack exec`). Remember that `stack exec` must be launched from the same directory that you ran `stack build` from.
```

Comment:
Should we drop this entirely and instead refer people to docs on that command?

Reply:
I don't think this usage info is described effectively in the [stack docs](https://docs.haskellstack.org/en/stable/GUIDE/#stack-exec).

Reply:
Hm... It does mention some part of the idea [here](https://docs.haskellstack.org/en/stable/GUIDE/#exec), but it's still not as clear as what you've written. 

I'd rather fix this upstream. Would you be up for submitting a PR to that repo to update the docs to clarify that? Then we don't have to explain it here when it should really be explained there.

> Remember that `stack exec` must be launched from the same directory that you ran `stack build` from.

I don't think that's quite true. I think it works as long as you are in a subdirectory of the root folder that contains the stack project.

Reply:
Or perhaps we just say

> f you're curious how `stack exec bash` works, run `echo $PATH` then `stack exec bash` then `echo $PATH` again, then `exit`, and finally `echo $PATH` a final time.)

Reply:
> Would you be up for submitting a PR to that repo to update the docs to clarify that?

I made a minor PR to the `stack` docs in https://github.com/commercialhaskell/stack/pull/5574

I'll leave out these two additional `stack exec bash` tips from this guide, and trust that users can figure this out on their own through trial and error:
- You must call `stack exec` from somewhere within the project directory
- You only need to call `stack exec` once

I added some additional tips (e.g. unix `which`) on how to determine which `purs` version is actually being run`.

Reply:
Thanks. Yeah, using `which` is a better approach than `stack path --local-bin`.